tests/Unit/IO/H5/Test_VolumeData.cpp

```diff
@@ -33,6 +35,198 @@ T multiply(const double obs_value, const T& component) {
+                grid_positions[j]) ==
+            multiply(observation_value,
+                     tensor_components_and_coords[grid_data_orders[j][i]]));
+    }
+  }
+}
+
+void test_strahlkorper() {
+  constexpr size_t l_max = 12;
+  constexpr size_t m_max = 12;
+  constexpr double sphere_radius = 4.0;
+  constexpr std::array<double, 3> center{{5.0, 6.0, 7.0}};
+  const Strahlkorper<Frame::Inertial> strahlkorper{l_max, m_max, sphere_radius,
+                                                   center};
+  const YlmSpherepack& ylm = strahlkorper.ylm_spherepack();
+  const std::array<DataVector, 2> theta_phi = ylm.theta_phi_points();
+  const DataVector theta = theta_phi[0];
+  const DataVector phi = theta_phi[1];
+  const DataVector sin_theta = sin(theta);
+  const DataVector radius = ylm.spec_to_phys(strahlkorper.coefficients());
+  const std::string grid_name{"AhA"};
+  const std::vector<DataVector> tensor_and_coord_data{
+      radius * sin_theta * cos(phi), radius * sin_theta * sin(phi),
+      radius * cos(theta), cos(2.0 * theta)};
+  const std::vector<TensorComponent> tensor_components{
+      {grid_name + "/InertialCoordinates_x", tensor_and_coord_data[0]},
+      {grid_name + "/InertialCoordinates_y", tensor_and_coord_data[1]},
+      {grid_name + "/InertialCoordinates_z", tensor_and_coord_data[2]},
+      {grid_name + "/TestScalar", tensor_and_coord_data[3]}};
+
+  const std::vector<size_t> observation_ids{4444};
+  const std::vector<double> observation_values{1.0};
+  const std::vector<Spectral::Basis> bases{2,
+                                           Spectral::Basis::SphericalHarmonic};
+  const std::vector<Spectral::Quadrature> quadratures{
+      {Spectral::Quadrature::Gauss, Spectral::Quadrature::Equiangular}};
+
+  const std::string h5_file_name{"Unit.IO.H5.VolumeData.Strahlkorper.h5"};
+  const uint32_t version_number = 4;
+  if (file_system::check_if_file_exists(h5_file_name)) {
+    file_system::rm(h5_file_name, true);
+  }
+
+  const std::vector<size_t> extents{
+      {ylm.physical_extents()[0], ylm.physical_extents()[1]}};
+
+  {
+    h5::H5File<h5::AccessType::ReadWrite> strahlkorper_file{h5_file_name};
+    auto& volume_file = strahlkorper_file.insert<h5::VolumeData>(
+        "/element_data", version_number);
+    volume_file.write_volume_data(
+        observation_ids[0], observation_values[0],
+        std::vector<ElementVolumeData>{
+            {extents, tensor_components, bases, quadratures}});
+
+    // Open the read volume file and check that the observation id and values
+    // are correct.
+    const auto& volume_file_read =
+        strahlkorper_file.get<h5::VolumeData>("/element_data", version_number);
+    const auto read_observation_ids = volume_file_read.list_observation_ids();
+    CHECK(read_observation_ids == std::vector<size_t>{4444});
+    CHECK(volume_file_read.get_observation_value(observation_ids[0]) ==
+          observation_values[0]);
+    }
```

Comment:
Is the indentation correct here? I'm confused, isn't this closing the brace on 199?